src/DeviceInterfaceValues.tsx

```diff
@@ -184,6 +189,346 @@ const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElemen
+          setErrors({ value: err.message });
+        });
+    }
+    if (interfaceData?.aggregation === 'object') {
+      const endpoints: string[] = interfaceData.mappings.map((mapping) => mapping.endpoint);
+      const endpointParts: string[][] = endpoints.map((endpoint) => endpoint.split('/'));
+      const commonParts: string[] = [];
+      for (let i = 0; i < endpointParts[0].length; i++) {
+        const currentPart: string = endpointParts[0][i];
+        if (endpointParts.every((parts) => parts[i] === currentPart)) {
+          commonParts.push(currentPart);
+        } else {
+          break;
+        }
+      }
+      setEndpoint(commonParts.join('/'));
+
+      interfaceData.mappings.forEach((mapping) => {
+        const path: string = mapping.endpoint.split('/').pop() || '';
+        data[path] = '';
+      });
+    }
+  }, [value, pathParams, paramValues]);
+
+  return (
+    <Modal size="lg" centered show={showModal} onHide={handleShowModal}>
+      <Form onSubmit={handleSubmit}>
+        <Modal.Header closeButton>
+          <Modal.Title>Publish data to interface</Modal.Title>
+        </Modal.Header>
+
+        <Modal.Body>
+          {(interfaceData?.aggregation === 'individual' ||
+            interfaceData?.type === 'properties') && (
+            <Form.Group as={Col} controlId="formEndpoint" className="mb-3">
+              <Form.Select as="select" value={endpoint} onChange={handleSelectedMapping}>
+                <option value="">Choose an endpoint for sending data</option>
+                {interfaceData?.mappings.map((mapping, index) => (
+                  <option key={index} value={mapping.endpoint}>
+                    {mapping.endpoint}
+                  </option>
+                ))}
+              </Form.Select>
+            </Form.Group>
+          )}
+
+          {!!pathParams.length && (
+            <Row className="d-flex justify-content-start my-4">
+              <p className="m-0">Please enter endpoint parameters:</p>
+              {pathParams.map((param, index) => (
+                <Col key={index} md="4" className="my-2">
+                  <Form.Group controlId={`param_${index}`}>
+                    <Form.Control
+                      type="text"
+                      required
+                      placeholder={param}
+                      value={paramValues[param] || ''}
+                      onChange={(e) => handleParamChange(param, e.target.value)}
+                      isInvalid={!paramValues[param]}
+                    />
+                  </Form.Group>
+                </Col>
+              ))}
+            </Row>
+          )}
+
+          {selectedMapping && (
+            <Form.Group as={Col} controlId="formValue">
+              <Form.Label className="m-0">Please enter the value:</Form.Label>
+              <Form.Control
+                type={
+                  selectedMapping.type === 'integer' ||
+                  selectedMapping.type === 'longinteger' ||
```

Comment:
careful with long integers.
When converting a longinteger into JavaScript numbers. Behind the scene every JavaScript number is a floating point, meaning you can lose precision when representing big numbers (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER for more details).
That's also why Astarte accepts longintegers as string.

Reply:
Thanks for advice, I have solved this through field validation, so now values greater than `max_safe_integer` can't be entered.

Reply:
but they need to be entered as they are supported by Astarte!
The difference is that they need to be sent as string. If we don't deal with that in this PR, let's add a TODO comment so we can fix it in a following PR